source/particles/particle_handler.cc

```diff
@@ -351,7 +376,70 @@ namespace Particles
   ParticleHandler<dim, spacedim>::remove_particle(
     const ParticleHandler<dim, spacedim>::particle_iterator &particle)
   {
-    particles.erase(particle->particle);
+    const unsigned int active_cell_index = particle->active_cell_index;
+
+    if (particles[active_cell_index].size() > 1)
```

Comment:
Shouldn't we check here that the cell is not ghost?

Reply:
You are right. Even better, since the accessor stores the point to the container it acts upon we should use that directly. I guess nobody tried removing a ghost particle individually so far (it is not something you would usually do).